### PR TITLE
Fix and order includes, using directives, makefile

### PR DESCRIPTION
**Pull request by dexX7**
There were many missing, but also many needless includes, as well as too wide imports of namespaces, which can lead to ambiguity.

Missing dependencies make it difficult to use Master Core in a different context, for example, and especially, in unit tests - which was by the way the reason I started with those maybe silly one-method-files like convert and parse_string, ...

Passes all tests:
- Boost
- [BitcoindComparisionTool](https://github.com/TheBlueMatt/test-scripts/tree/38b490a2599d422b12d5ce8f165792f63fd8f54f)
- [mastercore-rpc-tests](https://github.com/dexX7/mastercore-rpc-tests/tree/ebefe5d7a9a0788c96f3faa2debfc9880c2b5346)
- [bitcoin-spock](https://github.com/msgilligan/bitcoin-spock/tree/9a19a3bda7626d7719b58713181b2f49710175cc) ([test report](https://circle-artifacts.com/gh/dexX7/mastercore/32/artifacts/0/tmp/circle-artifacts.gvG7P9g/reports/tests/index.html))

Build log and results:
- [Circle CI](https://circleci.com/gh/dexX7/mastercore/32)
- [Travis CI](https://travis-ci.org/dexX7/mastercore/builds/48979402)

Basically nothing else than the dependency sections were touched.

### CODE REVIEW

**Comment by zathras-crypto** [comment]
pinging @m21 any objections here?  Safety level?

Thanks
Z

**Comment by dexX7** [comment]
To review:
1. Check, if no other code besides includes was changed.
2. Check, if it builds without error.

Since it touches the includes and usings, a failure should show up during the build process.

Two functional changes:

In src/mastercore_rpc.cpp, src/mastercore_rpc.h:

``` diff
-int populateRPCTransactionObject(uint256 txid, Object *txobj, string filterAddress = "")
+int populateRPCTransactionObject(const uint256& txid, Object *txobj, const string& filterAddress = "")
```

Instead of copying `txid` and `filterAddress`, a reference is passed.

In src/qt/transactiontablemodel.cpp:

``` diff
-extern CWallet* pwalletMain;
```

`pwalletMain` was not used there anywhere.

**Comment by dexX7** [comment]
Became part of https://github.com/mastercoin-MSC/mastercore/issues/286#issuecomment-77524801.
